articles/build-a-jekyll-site/index.md

```diff
@@ -0,0 +1,358 @@
+
+```jinja
+{% if page.title%}
+	<h1>{{page.title}}</h1>
+{% else %}
+	<h1>Untitled post</h1>
+{% endif %}
+{{ page.date | date: "%b %d, %Y" }}
+{{ content }}
+```
+
+Reload the page to see the changes.
+
+Jekyll uses Liquid template engine to generate the HTML files. You can learn more about Liquid tags and filters [here](https://shopify.github.io/liquid/).
+
+### Step 6 -- Adding images to posts
+To use images in your posts, create an `assets` folder in your base directory. Create an `images` folder inside `assets` and another `welcome` folder inside `images`. We are going to add a hero image and another image to your post.
+
+```bash
+$ mkdir -p assets/images/welcome
+```
+
+Copy 2 images to this directory and name one `hero.jpg` and the other `new-site.jpg`. You can download free stock photos from [Unsplash](https://unsplash.com/). The folder structure looks like this:
+
+```
+├── assets
+│   └── images
+│       └── welcome
+│           └── hero.jpg
+|           └── new-site.jpg
+```
+
+Let's add the hero image to your post. Add a `hero` property at the bottom of your front matter in your post markdown file.
+
+```yaml
+hero: /assets/images/welcome/hero.jpg
+```
+
+Add the `new-site.jpg` image at the bottom of your file.
+
+```markdown
+![demo image](/assets/images/welcome/new-site.jpg)
+```
+
+In your post layout file, add the hero image at the top.
+
+```jinja
+{% if page.hero %}
+	<img src="{{page.hero}}">
+{% endif %}
+```
+
+Save and reload the page. The hero image is displayed at the top and the other image appears at the bottom.
+
+### Step 7 -- Add navigation to your posts
+Up to this stage, your post doesn't have a navbar or a footer. These parts are an essential part of your site and should be on every page.
+
+Create an `_includes` folder in your base directory and create a `navbar.html` file inside. Put the following content inside it.
+
+```html
+<style>
+ul {
+  list-style-type: none;
+  display: flex;
+}
+li{
+ margin: 5px;
+}
+</style>
+
+<ul>
+  <li><a href="/">Home</a></li>
+  <li><a href="/about">About</a></li>
+</ul>
+```
+
+And then include your navbar at the top of your post layout file.
+
+```jinja
+{% include navbar.html %}
+```
+
+Save the files and reload the page to see the changes. The simple navbar in the `navbar.html` file will now be shown in each file that adopts the post layout. You can add the footer the same way.
+
+### Step 8 -- Adding custom styling to your blog
+Up to this stage, you are using the default `minima` theme provided by Jekyll. To style the site from scratch, delete the `theme` property in `_config.yml`.
+
+If you reload your site, it probably looks broken right now. Let's go ahead and rectify that. Create the following files under `_layouts`.
+
+- `default.html`
+- `page.html`
+- `home.html`
+
+Put the following content inside the files.
+
+\# default.html
```

Comment:
You should either remove the \ and have these as Markdown headings or include the page titles as comments inside the code snippets.